packages/rum/src/domain/record/mutationObserver.ts

```diff
@@ -98,15 +109,17 @@ export class MutationController {
 function processMutations(
   mutations: RumMutationRecord[],
   mutationCallback: MutationCallBack,
-  configuration: RumConfiguration
+  configuration: RumConfiguration,
+  shadowDomCreatedCb: ShadowDomCallBack,
+  target: Node
 ) {
   // Discard any mutation with a 'target' node that:
   // * isn't injected in the current document or isn't known/serialized yet: those nodes are likely
   // part of a mutation occurring in a parent Node
   // * should be hidden or ignored
   const filteredMutations = mutations.filter(
     (mutation): mutation is WithSerializedTarget<RumMutationRecord> =>
-      document.contains(mutation.target) &&
+      target.contains(mutation.target) &&
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏I think that when `mutation.target` is a Shadow Root, we should use its host instead.

Reply:
It works with the `shadowRoot`, Why do you want to use the host instead?